aiida/tools/data/orbital/realhydrogen.py

```diff
@@ -153,86 +321,53 @@ def _validate_keys(self, input_dict):
-
-        if validated_dict["angular_momentum"] is None:
-            raise InputValidationError("Must supply angular_momentum")
-        if validated_dict["magnetic_number"] is None:
-            raise InputValidationError("Must supply magnetic_number")
-        if validated_dict["radial_nodes"] is None:
-            validated_dict["radial_nodes"] = 0
+            accepted_range = [0, -angular_momentum]
+        if magnetic_number < min(accepted_range) or magnetic_number > max(accepted_range):
+            raise ValidationError("the magnetic number must be in the range [{}, {}]".format(
+                min(accepted_range), max(accepted_range)))
+
+        # Check if it is a known combination
         try:
             self.get_name_from_quantum_numbers(
-                validated_dict['angular_momentum'],
-                magnetic_number=validated_dict['magnetic_number'])
-        except InputValidationError:
-            raise InputValidationError("Invalid angular momentum magnetic"
-                                       " number combination.")
-
-        # Finally checks optional fields
-        KindName = input_dict.pop('kind_name', None)
-        if KindName:
-            if not isinstance(KindName, (six.string_types, None)):
-                raise ValidationError('If kind_name is provided must be string')
-            validated_dict['kind_name'] = KindName
-        Kind_index = input_dict.pop('kind_index', None)
-        if Kind_index:
-            if not isinstance(Kind_index, (int, None)):
-                raise ValidationError('If kind_index is provided must be int')
-            validated_dict['kind_index'] = Kind_index
-
-        if input_dict:
-            raise ValidationError("Some unrecognized keys: {}".
-                                  format(input_dict.keys()))
+                validated_dict['angular_momentum'], magnetic_number=validated_dict['magnetic_number'])
+        except ValidationError:
+            raise ValidationError("Invalid angular momentum magnetic number combination")
+
         return validated_dict
 
     @classmethod
-    def get_name_from_quantum_numbers(cls, angular_momentum,
-                                      magnetic_number=None):
+    def get_name_from_quantum_numbers(cls, angular_momentum, magnetic_number=None):
         """
         Returns the orbital_name correponding to the angular_momentum alone,
         or to both angular_number with magnetic_number. For example using
         angular_momentum=1 and magnetic_number=1 will return "Px"
         """
-        # importing a copy of the conversion_dict
-        convert_ref = copy.deepcopy(conversion_dict)
-        orbital_name = [x for x in convert_ref if
-                any([convert_ref[x][y]['angular_momentum'] ==
-                angular_momentum for y in convert_ref[x]])]
-        if len(orbital_name) == 0:
+        # importing a copy of the CONVERSION_DICT
+        convert_ref = copy.deepcopy(CONVERSION_DICT)
```

Comment:
You don't seem to be mutating the `convert_ref` so do you really need a deep copy here?

Reply:
indeed - I just left the old logic but just renamed to have pylint happy - now I fixed this